### ISSUE TITLE
When a 'model' is selected for sending an email, any extra attached file is deleted.

### ISSUE THREAD

**Issue by mdeweerd** [Dolibarr/dolibarr#16792]
# Bug
When a 'model' is selected for sending an email, any extra attached file is deleted.

Therefore sending emails from dolibarr with personalized attachments while using a model is impossible.

## Environment
- **Version**: 13.0.1
- **OS**: debian
- **PHP**: 7.3.X

## Expected and actual behavior
It is expected that files that are uploaded to be attached to the mail are actually attached even if it is an email based on a model.

## Steps to reproduce the behavior

1. Select an email model or define a default one so that a model is selected.
2. Upload a file to attach to the email.
3. Observe that the system reports that the upload is ok, but the uploaded file is actually not shown. It is not sent either.

## Information

I identified this by adding some dol_syslog code in `./core/class/html.formmail.class.php`
```php
$before=GETPOST('modelmailselected', 'alpha');
                // Clear temp files. Must be done at beginning, before call of triggers
                if (GETPOST('mode', 'alpha') === 'init' || (GETPOST('modelmailselected', 'alpha') && GETPOST('modelmailselected', 'alpha') != '-1'))
                {

 dol_syslog("getform/clear_attached files modeselectbefore=$before mode=".GETPOST('mode','alpha')." modelmailselected=".GETPOST('modelmailselected','alpha'));
                        $this->clear_attached_files();
                }

```
I changed the `== 'init'` test to `=== 'init'` to be more precise.

Which led to the following trace allowing to discover that the model was active, and that in this case the attached files are deleted.
```log
2021-03-18 11:07:00 INFO    10.0.94.52      getform/clear_attached files modeselectbefore=8 mode= modelmailselected=8
2021-03-18 11:07:00 DEBUG   10.0.94.52      functions.lib:dol_delete_dir_recursive /var/lib/dolibarr/documents/users/2/temp/
2021-03-18 11:07:00 INFO    10.0.94.52      dol_delete_file file=/var/lib/dolibarr/documents/users/2/temp//test.pdf disableglob=1 nophperrors=0 nohook=0
2021-03-18 11:07:00 DEBUG   10.0.94.52      Removed file /var/lib/dolibarr/documents/users/2/temp//test.pdf
```

Note that the path to the deleted file has a double slash, which could be improved.

To fix the issue for the person in trouble, I changed a line in `./core/class/html.formmail.class.php` to the next one, disabling deletion because a model was selected .
```php
                if (GETPOST('mode', 'alpha') === 'init' ||  (0 && GETPOST('modelmailselected', 'alpha') && GETPOST('modelmailselected', 'alpha') != '-1'))
```

With that code change, it was possible to actually attach a file to the mail - a test confirmed that the attached file was also received.

**Comment by daraelmin** [comment]
Duplicate #16423 and #16420

Try to upgrade to 13.0.2

**Comment by daraelmin** [comment]
@mdeweerd if you update your dolibarr to 13.0.2, you may need to read this thread #16767 and use this fix #16884 
Cheers

**Comment by mdeweerd** [comment]
@daraelmin The setup was updated to 13.0.2 and it's possible to add attachments.
I'll keep in mind that there are some other limitations corrected by fixes (that possibly become available in 13.0.3).

Thank you for the warning.

**Comment by daraelmin** [comment]
@mdeweerd with pleasure
